libpod/runtime_img.go

```diff
@@ -123,41 +211,107 @@ func (r *Runtime) PullImage(imgName string, allTags bool, signaturePolicyPath st
 		tarSource := tarfile.NewSource(archFile)
 		manifest, err := tarSource.LoadTarManifest()
 		if err != nil {
-			return errors.Errorf("error retrieving manifest.json: %v", err)
+			return nil, errors.Errorf("error retrieving manifest.json: %v", err)
 		}
-		// to pull all the images stored in one tar file
-		for i := range manifest {
-			if manifest[i].RepoTags != nil {
-				images = append(images, manifest[i].RepoTags[0])
-			} else {
-				// create an image object and use the hex value of the digest as the image ID
-				// for parsing the store reference
-				newImg, err := srcRef.NewImage(sc)
+		// to pull the first image stored in the tar file
+		if len(manifest) == 0 {
+			// create an image object and use the hex value of the digest as the image ID
+			// for parsing the store reference
+			newImg, err := srcRef.NewImage(sc)
```

Comment:
`len(manifest) == `**`0`**? That means there isn’t any image to pull at all (and `NewImage` should always fail), doesn’t it?